### PR TITLE
[SPARK-27531][SQL] Improve `EXPLAIN DESC TABLE` to show the input parameters of the command.

### PR DESCRIPTION
**Pull request by dilipbiswal**
## What changes were proposed in this pull request?
Currently "EXPLAIN DESC TABLE" is special cased and outputs a single row relation as following. 
Current output:
```sql
spark-sql> EXPLAIN DESCRIBE TABLE t;
== Physical Plan ==
*(1) Scan OneRowRelation[]
```
This is not consistent with how we handle explain processing for other commands. In this PR, the inconsistency is handled by removing the special handling for "describe table".

After change:
```sql
spark-sql> EXPLAIN DESC EXTENDED t
== Physical Plan ==
Execute DescribeTableCommand
   +- DescribeTableCommand `t`, true
```
## How was this patch tested?
Added new tests in SQLQueryTestSuite.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala [hunk 1]
```diff
@@ -318,25 +318,15 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
...
-    } else {
-      ExplainCommand(OneRowRelation())
     }
   }
 
-  /**
-   * Determine if a plan should be explained at all.
-   */
-  protected def isExplainableStatement(plan: LogicalPlan): Boolean = plan match {
-    case _: DescribeTableCommand => false
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala]: Hi, @hvanhovell .
This was introduced at Spark 2.0.0 by SPARK-13713 . Can we remove this at 3.0.0?

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
Thank you, @dilipbiswal .

**Comment by dilipbiswal** [comment]
Thanks a LOT @dongjoon-hyun

**Pull request closed without merging**